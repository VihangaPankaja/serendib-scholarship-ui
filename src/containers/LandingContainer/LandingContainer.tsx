import React, { Component } from 'react';
import TableComponent from '../../components/TableComponent/TableComponent';
import { Navbar, Container, Nav, Button, Tab, Col, Row } from 'react-bootstrap';
import { QueryClient, QueryClientProvider } from "react-query";
import logo from '../../assets/images/logo1.png';

const queryClient = new QueryClient();

interface LandingContainerProp { }

const btn = { backgroundColor: '#212529' };

export default class LandingContainer extends Component<LandingContainerProp> {
    constructor(props: LandingContainerProp) {
        super(props);
    }

    render(): React.ReactNode {
        return <>
            <Navbar bg="dark" variant="dark" fixed="top" style={{height: '70px'}}>
                <Container fluid>
                    <Navbar.Brand style={{backgroundColor: 'black', borderRadius: '10px'}}><span style={{margin: '10px'}}>Serendib Foundation</span></Navbar.Brand>
                    <Nav className="ml-auto">
                        <Nav.Link href="http://credoxyz.epizy.com/?i=1"><img style={{width: '50px', height: '50px', borderRadius: '50px'}} src={logo} /></Nav.Link>
                    </Nav>
                </Container>
            </Navbar>
            <Container style={{...btn, top: '70px', position: 'relative'}} fluid>
                <Tab.Container id="left-tabs-example" defaultActiveKey="first">
                    <Row style={{backgroundColor: '#212121'}}>
                        <Col sm={1} style={{backgroundColor: 'black', width: '150px', height: '100%', position: 'fixed'}}>
                            <Nav style={{marginTop: '15px'}} variant="pills" className="flex-column">
                                <Nav.Item>
                                    <Nav.Link eventKey="first" style={{textAlign: 'center'}}>Student</Nav.Link>
                                </Nav.Item>
                                <Nav.Item  style={{marginTop: '15px'}}>
                                    <Nav.Link eventKey="second" style={{textAlign: 'center'}}>Sponsors</Nav.Link>
                                </Nav.Item>
                            </Nav>
                        </Col>
                        <Col sm={11} style={{marginLeft: '150px', position: 'relative'}}>
                            <QueryClientProvider client={queryClient}>
                                <Tab.Content>
                                    <Tab.Pane eventKey="first">
                                        <TableComponent type={'student'} />
                                    </Tab.Pane>
                                    <Tab.Pane eventKey="second">
                                        <TableComponent type={'sponsor'} />
                                    </Tab.Pane>
                                </Tab.Content>
                            </QueryClientProvider>
                        </Col>
                    </Row>
                </Tab.Container>
            </Container>
        </>
    }
}