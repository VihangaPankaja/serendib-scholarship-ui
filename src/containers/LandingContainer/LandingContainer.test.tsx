import { render } from "@testing-library/react";
import React from "react";
import { describe, it, expect } from "@jest/globals";
import LandingContainer from "./LandingContainer";

describe("LandingContainer", () => {
  it("renders", () => {
    const wrapper = render(<LandingContainer />);
    expect(wrapper.container).toMatchSnapshot();
  });
});