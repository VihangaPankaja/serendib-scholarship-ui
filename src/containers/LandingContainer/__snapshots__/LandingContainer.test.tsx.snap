// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingContainer renders 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      class="container-fluid"
    >
      <span
        class="navbar-brand"
      >
        Serendib Foundation
      </span>
    </div>
  </nav>
  <div
    class="container-fluid"
    style="background-color: rgb(33, 37, 41);"
  >
    <div
      class="row"
      style="background-color: rgb(33, 37, 41);"
    >
      <div
        class="col-sm-1"
      >
        <div
          class="flex-column nav nav-pills"
          role="tablist"
        >
          <div
            class="nav-item"
          >
            <a
              aria-controls="left-tabs-example-tabpane-first"
              aria-selected="true"
              class="nav-link active"
              data-rr-ui-event-key="first"
              href="#"
              id="left-tabs-example-tab-first"
              role="tab"
              tabindex="0"
            >
              Student
            </a>
          </div>
          <div
            class="nav-item"
          >
            <a
              aria-controls="left-tabs-example-tabpane-second"
              aria-selected="false"
              class="nav-link"
              data-rr-ui-event-key="second"
              href="#"
              id="left-tabs-example-tab-second"
              role="tab"
              tabindex="-1"
            >
              Sponsors
            </a>
          </div>
        </div>
      </div>
      <div
        class="col-sm-11"
      >
        <div
          class="tab-content"
        >
          <div
            aria-labelledby="left-tabs-example-tab-first"
            class="fade tab-pane active show"
            id="left-tabs-example-tabpane-first"
            role="tabpanel"
          >
            <div
              aria-colcount="10"
              aria-rowcount="3"
              class="rdg r104f42s700-beta15"
              dir="ltr"
              role="grid"
              style="height: 100vh; grid-template-rows: 35px repeat(2, 35px); --rdg-header-row-height: 35px; --rdg-summary-row-height: 35px; --rdg-sign: 1; grid-template-columns: 80px 80px 80px 80px 80px 80px 80px 80px 80px 80px;"
            >
              <div
                aria-rowindex="1"
                class="rdg-header-row h197vzie700-beta15"
                role="row"
                style="--rdg-grid-row-start: 1;"
              >
                <div
                  aria-colindex="1"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="columnheader"
                  style="grid-column-start: 1; min-width: 80px;"
                  tabindex="0"
                >
                  ID
                </div>
                <div
                  aria-colindex="2"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="columnheader"
                  style="grid-column-start: 2; min-width: 80px;"
                  tabindex="-1"
                >
                  Name
                </div>
              </div>
              <div
                aria-rowindex="2"
                class="rdg-row r1otpg64700-beta15 rdg-row-even"
                role="row"
                style="--rdg-grid-row-start: 2; --rdg-row-height: 35px;"
              >
                <div
                  aria-colindex="1"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 1;"
                  tabindex="-1"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    0
                  </button>
                </div>
                <div
                  aria-colindex="2"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 2;"
                  tabindex="-1"
                >
                  Example
                </div>
              </div>
              <div
                aria-rowindex="3"
                class="rdg-row r1otpg64700-beta15 rdg-row-odd"
                role="row"
                style="--rdg-grid-row-start: 3; --rdg-row-height: 35px;"
              >
                <div
                  aria-colindex="1"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 1;"
                  tabindex="-1"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    1
                  </button>
                </div>
                <div
                  aria-colindex="2"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 2;"
                  tabindex="-1"
                >
                  Demo
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="left-tabs-example-tab-second"
            class="fade tab-pane"
            id="left-tabs-example-tabpane-second"
            role="tabpanel"
          >
            <div
              aria-colcount="2"
              aria-rowcount="3"
              class="rdg r104f42s700-beta15"
              dir="ltr"
              role="grid"
              style="height: 100vh; grid-template-rows: 35px repeat(2, 35px); --rdg-header-row-height: 35px; --rdg-summary-row-height: 35px; --rdg-sign: 1; grid-template-columns: 80px 80px;"
            >
              <div
                aria-rowindex="1"
                class="rdg-header-row h197vzie700-beta15"
                role="row"
                style="--rdg-grid-row-start: 1;"
              >
                <div
                  aria-colindex="1"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="columnheader"
                  style="grid-column-start: 1; min-width: 80px;"
                  tabindex="0"
                >
                  ID
                </div>
                <div
                  aria-colindex="2"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="columnheader"
                  style="grid-column-start: 2; min-width: 80px;"
                  tabindex="-1"
                >
                  Name
                </div>
              </div>
              <div
                aria-rowindex="2"
                class="rdg-row r1otpg64700-beta15 rdg-row-even"
                role="row"
                style="--rdg-grid-row-start: 2; --rdg-row-height: 35px;"
              >
                <div
                  aria-colindex="1"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 1;"
                  tabindex="-1"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    0
                  </button>
                </div>
                <div
                  aria-colindex="2"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 2;"
                  tabindex="-1"
                >
                  Example
                </div>
              </div>
              <div
                aria-rowindex="3"
                class="rdg-row r1otpg64700-beta15 rdg-row-odd"
                role="row"
                style="--rdg-grid-row-start: 3; --rdg-row-height: 35px;"
              >
                <div
                  aria-colindex="1"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 1;"
                  tabindex="-1"
                >
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    1
                  </button>
                </div>
                <div
                  aria-colindex="2"
                  aria-readonly="true"
                  aria-selected="false"
                  class="rdg-cell c1wupbe700-beta15"
                  role="gridcell"
                  style="grid-column-start: 2;"
                  tabindex="-1"
                >
                  Demo
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
